lib/gui/app/components/settings/settings.jsx

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const React = require('react')
+const { useState } = React
+const propTypes = require('prop-types')
+const _ = require('lodash')
+const store = require('../../models/store')
+const analytics = require('../../modules/analytics')
+const settings = require('../../models/settings')
+const { default: styled } = require('styled-components')
+const { FaCog } = require('react-icons/fa')
+const {
+  Badge,
+  Button,
+  Checkbox,
+  Modal,
+  Provider
+} = require('rendition')
+const { colors } = require('../../theme')
+
+const SettingsIcon = styled(FaCog) `
+  width: 24px;
+  height: 24px;
+
+  &&& {
+    color: ${colors.secondary.background}!important;
+  }
+`
+
+const SettingsButton = () => {
+  const [ hideModal, setHideModal ] = useState(true)
+
+  return (
+    <Provider>
+      <Button
+        icon={<SettingsIcon/>}
+        plain
+        onClick={() => setHideModal(false)}
+        tabIndex="5">
+      </Button>
+      { hideModal ? null : (
+        <SettingsModal toggleModal={(value) => setHideModal(!value)}>
+        </SettingsModal>
+      ) }
+    </Provider>
+  )
+}
+
+SettingsButton.propTypes = {}
+
+const WarningModal = ({
+  message,
+  confirmLabel,
+  cancel,
+  done
+}) => {
+  return (
+    <Modal
+      title={confirmLabel}
+      action={confirmLabel}
+      cancel={cancel}
+      done={done}
+      style={{
+        width: 420,
+        height: 300
+      }}
+      primaryButtonProps={{ warning: true }}>
+      {message}
+    </Modal>
+  )
+}
+
+const SettingsModal = styled((props) => {
+  const [ currentSettings, setCurrentSettings ] = useState(settings.getAll())
```

Comment:
This logic doesn't work if anything outside this file modifies the settings: `currentSettings` and the actual settings will diverge.
We should always be showing the settings from `settings.getAll()`

Reply:
We don't have an observer on the settings right now, as they're not meant to be changed outside of the settings page afaict. We'd have to poll them repeatedly

Reply:
Ok, maybe leave a comment here.
If not polling repeatedly, we could at least reload them all when one is changed.